src/System/src/FloatingBaseSystemDynamics.cpp

```diff
@@ -142,29 +141,17 @@ bool FloatingBaseDynamicalSystem::dynamics(const double& time,
     jointVelocityOutput = jointVelocity;
 
     // update kindyncomputations object
-    // convert the objects in iDynTree objects. This will solve as soon as the kinDyn object can
-    // take spans as input
-    iDynTree::Twist baseTwist;
-    iDynTree::toEigen(baseTwist.getLinearVec3()) = baseVelocity.head<3>();
-    iDynTree::toEigen(baseTwist.getAngularVec3()) = baseVelocity.tail<3>();
-
-    iDynTree::Rotation baseRot;
-    iDynTree::toEigen(baseRot) = baseOrientation;
-
-    iDynTree::Position basePos;
-    iDynTree::toEigen(basePos) = basePosition;
-
-    iDynTree::VectorDynSize jointPos(jointVelocityOutput.size());
-    iDynTree::toEigen(jointPos) = jointPositions;
-
-    iDynTree::VectorDynSize jointVel(jointVelocityOutput.size());
-    iDynTree::toEigen(jointVel) = jointVelocity;
-
-    if (!m_kinDyn->setRobotState(iDynTree::Transform(baseRot, basePos),
-                                 jointPos,
-                                 baseTwist,
-                                 jointVel,
-                                 m_gravity))
+    Eigen::Matrix4d baseTransform;
+    baseTransform.topLeftCorner<3,3>() = baseOrientation;
+    baseTransform.topRightCorner<3,1>() = basePosition;
+    baseTransform(3,3) = 1;
+    baseTransform.bottomLeftCorner<1,3>().setZero();
```

Comment:
We could maybe add a utility method for this somewhere?

Reply:
Done here: https://github.com/dic-iit/bipedal-locomotion-framework/pull/108/commits/27952e9fb75e543331dbcb6b2902da36579e07d3

Reply:
Please see https://github.com/dic-iit/bipedal-locomotion-framework/pull/108#discussion_r527050397